core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -615,6 +627,32 @@ private RelationPlan getInsertPlan(
                 statisticsMetadata);
     }
 
+    private RefreshType canIncrementallyRefresh(RelationPlan plan)
+    {
+        return new IncrementalRefreshVisitor().visitPlan(plan.getRoot(), null);
+    }
+
+    static class IncrementalRefreshVisitor
+            extends PlanVisitor<RefreshType, Void>
+    {
+        @Override
+        protected RefreshType visitPlan(PlanNode node, Void context)
+        {
+            if (node instanceof AggregationNode
```

Comment:
It most definitely should be white-listing for incremental refresh rather (so operators that allow for incremental refresh). New nodes might be added which would not support incremental refresh.

Reply:
Makes sense, changed this to white-listing approach